### PR TITLE
kotlin code cleanup

### PR DESCRIPTION
**Pull request by pratyaksh1610**
## Purpose / Description
Kotlin cleanup

## Approach
_How does this change address the problem?_

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration (SDK version(s), emulator or physical, etc)

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/services/BootService.kt [hunk 1]
```diff
@@ -114,9 +114,11 @@ class BootService : BroadcastReceiver() {
                 return
             }
             val calendar = time.calendar()
-            calendar[Calendar.HOUR_OF_DAY] = getRolloverHourOfDay(context)
-            calendar[Calendar.MINUTE] = 0
-            calendar[Calendar.SECOND] = 0
+            calendar.apply {
+                Calendar.HOUR_OF_DAY to getRolloverHourOfDay(context)
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/services/BootService.kt:118]: The code is not equivalent. Your code:

`Calendar.HOUR_OF_DAY to getRolloverHourOfDay(context)`

simply creates a Pair<Int, Int> inside the apply block not changing the calendar instance. You can see this as you can assign the pair above to a variable:

```
calendar.apply {
     val unused: Pair<Int, Int> = Calendar.HOUR_OF_DAY to getRolloverHourOfDay(context)
```

So please use the set() method inside the apply block as that is the method used when using the bracket style from the initial code.

File: AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt [hunk 2]
```diff
@@ -414,16 +415,12 @@ create table meta (dirMod int, lastUsn int); insert into meta values (0, 0);"""
      * Illegal characters and paths
      * ***********************************************************
      */
-    @KotlinCleanup("one line function")
     fun stripIllegal(str: String): String {
-        val m = fIllegalCharReg.matcher(str)
-        return m.replaceAll("")
+        return fIllegalCharReg.matcher(str).replaceAll("")
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt:419]: You can further improve this function(and the other one) with:

`fun stripIllegal(str: String): String = fIllegalCharReg.matcher(str).replaceAll("")`

**Review by lukstbit — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt [hunk 3]
```diff
@@ -939,9 +914,7 @@ create table meta (dirMod int, lastUsn int); insert into meta values (0, 0);"""
             fObjectRegExpU
         )
 
-        fun getCollectionMediaPath(collectionPath: String): String {
-            return collectionPath.replaceFirst("\\.anki2$".toRegex(), ".media")
-        }
+        fun getCollectionMediaPath(collectionPath: String): String = collectionPath.replaceFirst("\\.anki2$".toRegex(), ".media")
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt:917]: Please revert this change. While we can turn this into a one line function, it becomes a bit too long to read inside an IDE.

File: AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt [hunk 4]
```diff
@@ -979,9 +952,7 @@ create table meta (dirMod int, lastUsn int); insert into meta values (0, 0);"""
          * (Anki2Importer needs this). This is needed because we didn't implement the "transformNames"
          * function and have delegated its job to the caller of this class.
          */
-        fun indexOfFname(p: Pattern): Int {
-            return if (p == fSoundRegexps) 2 else if (p == fImgAudioRegExpU) 2 else 3
-        }
+        fun indexOfFname(p: Pattern): Int = if (p == fSoundRegexps) 2 else if (p == fImgAudioRegExpU) 2 else 3
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt:955]: Please revert this change. It becomes a bit too long and it's also more difficult to read with the nested if-else.

**Review by lukstbit — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt (hunk 3 above)
Comment (pratyaksh1610) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt]: Done

File: AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt (hunk 4 above)
Comment (pratyaksh1610) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt]: Done

File: AnkiDroid/src/main/java/com/ichi2/anki/services/BootService.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (pratyaksh1610) [reply AnkiDroid/src/main/java/com/ichi2/anki/services/BootService.kt]: I have made the change. Thanks.

File: AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (pratyaksh1610) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt]: I have made the change. Thanks.

**Review by lukstbit — APPROVED** [review]

File: AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt [hunk 5]
```diff
@@ -139,9 +140,8 @@ create table meta (dirMod int, lastUsn int); insert into meta values (0, 0);"""
     }
 
     @KotlinCleanup("nullable if server == true, we don't do this in AnkiDroid so should be fine")
-    fun dir(): String {
-        return mDir!!
-    }
+    fun dir(): String = mDir!!
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Media.kt:143]: To be honest, this kind of change (removing `return` to use `=` on existing code) only generates noise and doesn't serve an useful purpose, IMO

**Review by BrayanDSO — APPROVED** [review]
Approving, but 80% of it wasn't really necessary

**Pull request merged**